agent/internal/client/testdata/pre_error.yaml

```diff
@@ -0,0 +1,33 @@
+command_type: system
+pre_error_fails_task: true
+
+pre:
+  - func: foo
+
+functions:
+  "foo":
+    - command: shell.exec
+      type: setup
+      params:
+        shell: bash
+        script: |
+          exit 1
```

Comment:
If it's a one-line shell script, you can just put it on the same line rather than using `|` to stick it on the next line.